sdk/network/mgmt/src/main/java/com/azure/management/network/implementation/VirtualNetworkGatewayImpl.java

```diff
@@ -414,6 +414,7 @@ VirtualNetworkGatewayIPConfiguration defaultIPConfiguration() {
             // ...and no need to create VNet
             networkObservable = Mono.empty(); // ...and don't create another VNet
         } else {
+            assert creatableNetwork != null;
```

Comment:
Not sure if it better use if instead of assert

Reply:
Maybe just check not null, throw IllegalState if null?

Reply:
done